### PR TITLE
Fix bug in duplicate distribution columns for CTAS

### PR DESCRIPTION
**Pull request by danielgustafsson**
When creating a table using the CTAS construction with an explicit
distribution clause, the duplicate check for distribution columns
wasn't applied:
```SQL
CREATE TABLE tt AS SELECT * FROM t DISTRIBUTED BY (c,c);
```
Move the duplicate distribution check to the parsing stage since
it is a syntax error to supply duplicate column references. This
alters the behavior such that supplying a duplicate non-existing
column will error out on the duplication rather than the column
being non-existing. Since syntax errors should have precedence
over semantic errors I however find that this is the correct order.

Remove the existing duplicate check which only applied to CREATE
TABLE and add test cases to cover CTAS. The ticket reference was
removed since it's not public and also clearly didn't cover all
the cases. The DROP TABLE is also moved to not using IF EXISTS
since it would be an error if the table didn't exist at that point.

This fixes issue #2120

### CODE REVIEW

**Comment by danielgustafsson** [comment]
For the INSERT targets I agree that it makes more sense to check for duplicates in parse analysis, in general though I think syntax errors should be caught as early as possible.

**Comment by danielgustafsson** [comment]
Pushed an update which removes the (duplicated) duplicate check from ALTER TABLE such that we only have one place for distribution duplicate elimination.

Also added a commit cleaning a little around where I had to touch the ALTER code as it was in some need of a little TLC.

**Comment by danielgustafsson** [comment]
(a previous version changed the `strcat()` calls to `strlcat()` but since the unittest compilation step requires some refactoring to handle that, I reverted that part rather than adding a kluge in the Makefiles. Will at some point fix the entire unittest compilation instead)

File: src/test/regress/expected/alter_distribution_policy.out [hunk 1]
```diff
@@ -13,8 +13,9 @@ LINE 1: alter table atsdb set distributed by ();
 alter table atsdb set distributed by (m);
 ERROR:  column "m" of relation "atsdb" does not exist
 alter table atsdb set distributed by (i, i);
-ERROR:  distribution policy must be a unique set of columns
-HINT:  Column "i" appears more than once
+ERROR:  duplicate column in DISTRIBUTED BY clause
```
Comment (vraghavan78) [inline src/test/regress/expected/alter_distribution_policy.out]: @mkiyama @dyozie can you please add documentation w.r.t change in user surface. Else we will have wrong documentation.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks

**Pull request closed without merging**